### DISCUSSION TITLE
Mandatory fields on RDFDataset

### DISCUSSION THREAD

**Discussion by miekeGR** [ternaustralia/ontology_tern#84] [Modelling]
Referring to the worked example: https://github.com/ternaustralia/bdr-faealla-worked-example/tree/5bf4b70f1b60ed47fff7a300a01062b62568e5ad I noticed that the RDFDataset nodes did not include dcterms:description, dcterms:issued, and dcterm:title, which are listed as requiring exactly 1 on  https://linkeddata-dev.tern.org.au/viewers/tern-ontology?uri=https://w3id.org/tern/ontologies/tern/RDFDataset. 

I was wondering if these fields are required in a implementation of the ontology at a metadata level or are they meant to come from the data submitter? I get the feeling that dcterms:issued is definitely an implementation type field. dcterms:title and dcterms:description being required from a submitter would be fairly tedious since the nature of the data content implies that information already in some way.

**Comment by edmondchuc** [comment]
Hi @miekeGR, I think RDFDataset in this case needs to be revised. Before the worked example, the RDFDataset class was meant to provide some metadata information for a payload of RDF data to an integration platform (such as the BDR). But through the worked example, it was realised that the RDFDataset class can also be used to capture information in place of a Darwin Core Record.

I agree some of the mandatory properties are tedious. Let's review these requirements. 

I'm also unfamiliar right now what is required in the BDR when you submit RDF data to the BDR gateway. Is RDFDataset even used here?

**Reply by miekeGR** [reply]
Cheers, thanks @edmondchuc I think RDFDataset is quite good at acting as the Darwin Core Record. 

While the specific requirements of which attributes get added to the RDFDataset during the submission process have not been formalised yet, I see dcterms:issued as being one that will be part of the system processing, not required beforehand.

Looking forward to chatting about this further at some stage.

**Reply by edmondchuc** [reply]
If the current mandatory properties are made optional, then we probably don't need RDFDataset anymore and can just use the original void:Dataset class [1]. On the other hand, having RDFDataset provides us the flexibility to add other mandatory or optional properties tailored towards the ecological domain. 

[1] https://www.w3.org/TR/void/#metadata

**Comment by dr-shorthair** [comment]
I think it's worth having an explicit container in our ontology. We could axiomatize an alignment, but as noted we will almost certainly want to add some axioms or constraints that would not apply to `void:Dataset`

**Reply by edmondchuc** [reply]
What should this `Record` class look like? And what relationships should it have?

Are we happy with the pattern we are using now for modelling Darwin Core Record? We currently define an `RDFDataset` instance and link each individual to the `RDFDataset` instance with the `void:inDataset` property. The `RDFDataset` instance has _incoming_ relationships.

Alternatively, the relationship can be from the `Record` to each individual, similar to `sosa:ObservationCollection` with its usage of `sosa:hasMember` property. The `Record` instance has _outgoing_ relationships.

**Comment by edmondchuc** [comment]
`RDFDataset` no longer have cardinality constraints. See https://w3id.org/tern/ontologies/tern/RDFDataset.
